src/components/Workflow/SearchableWorkflowNameList.tsx

```diff
@@ -1,135 +1,53 @@
 import { Typography } from '@material-ui/core';
-import { makeStyles, Theme } from '@material-ui/core/styles';
 import ChevronRight from '@material-ui/icons/ChevronRight';
-import { SearchableList, SearchResult } from 'components/common/SearchableList';
+import { SearchResult } from 'components/common';
+import {
+    SearchableNamedEntity,
+    SearchableNamedEntityList,
+    SearchableNamedEntityListProps,
+    useNamedEntityListStyles
+} from 'components/common/SearchableNamedEntityList';
 import { useCommonStyles } from 'components/common/styles';
-import { listhoverColor, separatorColor } from 'components/Theme';
-import { NamedEntity } from 'models';
 import * as React from 'react';
 import { Link } from 'react-router-dom';
 import { Routes } from 'routes';
 
-const useStyles = makeStyles((theme: Theme) => ({
-    container: {
-        marginBottom: theme.spacing(2),
-        width: '100%'
-    },
-    itemName: {
-        flex: '1 0 auto',
-        padding: `${theme.spacing(2)}px 0`
-    },
-    itemChevron: {
-        color: theme.palette.grey[500],
-        flex: '0 0 auto'
-    },
-    noResults: {
-        color: theme.palette.text.disabled,
-        display: 'flex',
-        justifyContent: 'center',
-        marginTop: theme.spacing(6)
-    },
-    searchResult: {
-        alignItems: 'center',
-        borderBottom: `1px solid ${separatorColor}`,
-        display: 'flex',
-        flexDirection: 'row',
-        padding: `0 ${theme.spacing(3)}px`,
-        '&:first-of-type': {
-            borderTop: `1px solid ${separatorColor}`
-        },
-        '&:hover': {
-            backgroundColor: listhoverColor
-        },
-        '& mark': {
-            backgroundColor: 'unset',
-            color: theme.palette.primary.main,
-            fontWeight: 'bold'
-        }
-    }
-}));
-
-interface SearchableWorkflowName extends NamedEntity {
-    key: string;
-}
-
-const workflowNameKey = ({ id: { domain, name, project } }: NamedEntity) =>
-    `${domain}/${name}/${project}`;
-
-const NoResults: React.FC = () => (
-    <Typography className={useStyles().noResults} variant="h6" component="div">
-        No matching results
-    </Typography>
-);
-
-interface SearchResultsProps {
-    results: SearchResult<SearchableWorkflowName>[];
-}
-const SearchResults: React.FC<SearchResultsProps> = ({ results }) => {
+/** Renders a searchable list of Workflow names, with associated descriptions */
+export const SearchableWorkflowNameList: React.FC<
+    Omit<SearchableNamedEntityListProps, 'renderItem'>
```

Comment:
If this type is common, it may be worth declaring someplace. Not a blocker, just noticing because I saw it twice.

Reply:
Ah, yes. I'm making additional changes to this stuff in a follow-up branch. I'll move it to a common place if it stays common.